### DISCUSSION TITLE
Migration from influxdb2: large increase in storage

### DISCUSSION THREAD

**Discussion by azertylr** [questdb/questdb#4988] [Show and tell]
Hello,

I've been using influxdb2 for IoT sensors data storage for several years and I wanted to try questdb that sounds promising!

I've used a python script to read my influxdb2 database and write (using the official influxdb python client), to questdb. I did migrate half of my db. It has 55M rows and 122 columns. The problem is that it uses 39 Go on my hard drive, vs 500 Mo with influxdb!

I only have 6 tags on influx db and around 100 fields. Most points only have 1 field. But it seems that questdb stores the all columns for each point?

I'm not sure how I can optimize it!

Thank!

**Comment by bluestreak01** [comment]
you can run QuestDB on ZFS

**Reply by azertylr** [reply]
going from 500 Mo to 39 Go (and I've stopped the transfert before the end), 100mo with VictoriaMetrics, seems strange to me 

is it normal (without ZFS) or is it a misuse of questdb?

**Comment by danisam** [comment]
hello @azertylr , did you solve your problem ? i am very interested in migrated from INfluxdb to QuestDB.

**Comment by azertylr** [comment]
Hi, 

I haven't tried since last september. I don't known if they make any improvements on this point.

I'm still using influxdb2 after tester questdb, victoriametric and others.

**Comment by bluestreak01** [comment]
QuestDB would not be an ideal solution for sparse data. If you arrange your data such that most of your columns are NULL - InfluxDB will work better. QuestDB will store nulls... Unless data partition is parquet, NULLs will be compressed.

**Comment by ideoma** [comment]
You can now use PARQUET format for the table. It compresses the data without ZFS
https://questdb.com/docs/query/sql/alter-table-set-format/#description
